File: my-submissions/m1940 counter.py
# weekly premium question

# Rank: 2

# Counts occurances of each number
# Since all values are STRICTLY increasing, repeats aren't present
# 2'd fastest it seems behind the subset method

class Solution:
    def longestCommonSubsequence(self, arrays: List[List[int]]) -> List[int]:
        counter = {}

        for subArr in arrays :
            for i in subArr :
                counter[i] = counter.get(i, 0) + 1
        
        return [x for x in counter.keys() if counter.get(x, 0) == len(arrays)]

File: my-submissions/m1940 counter.java
// weekly premium question

class Solution {
    public List<Integer> longestCommonSubsequence(int[][] arrays) {
        HashMap<Integer, Integer> counter = new HashMap<>();

        for (int[] subArr: arrays){
            for (int i: subArr) {
                counter.put(i, counter.getOrDefault(i, 0) + 1);
            }
        }

        ArrayList<Integer> output = new ArrayList<>();
        
        for (int i: counter.keySet()) {
            if (counter.getOrDefault(i, 0) == arrays.length) {
                output.add(i);
            }
        }
        Collections.sort(output);

        return output;
    }
}